src/playback-watcher.js

```diff
@@ -356,17 +370,22 @@ export default class PlaybackWatcher {
       return true;
     }
 
+    const sourceUpdater = this.masterPlaylistController_.sourceUpdater_;
     const buffered = this.tech_.buffered();
 
     if (
       closeToBufferedContent({
         buffered,
+        audioBuffered: sourceUpdater.audioBuffer ? sourceUpdater.audioBuffered() : null,
+        videoBuffered: sourceUpdater.videoBuffer ? sourceUpdater.videoBuffered() : null,
         targetDuration: this.media().targetDuration,
         currentTime
       })
     ) {
-      seekTo = buffered.start(0) + Ranges.SAFE_TIME_DELTA;
-      this.logger_(`Buffered region starts (${buffered.start(0)}) ` +
+      const nextRange = Ranges.findNextRange(buffered, currentTime);
+
+      seekTo = nextRange.start(0) + Ranges.SAFE_TIME_DELTA;
```

Comment:
There are three parts.
1. We have to consider all content (including gaps) after current time to determine if we have appended two segments. Previously we just used the first buffer to determine this. This is in `closeToBufferedContent`
2. Next we need to find the range that is ahead of currentTime, rather than the first buffer, and determine if it is within one segment duration away from currentTime. This is also in `closeToBufferedContent`
3. Finally we need to again get the `nextRange` rather than the first one to determine where we should seek.